File: markymark-android/src/main/java/com/m2mobi/markymarkandroid/inline/BoldInlineDisplayItem.java
/*
* Copyright (C) M2mobi BV - All Rights Reserved
*/

package com.m2mobi.markymarkandroid.inline;

import android.graphics.Typeface;
import android.text.Spannable;
import android.text.Spanned;
import android.text.style.StyleSpan;

import com.m2mobi.markymark.InlineConverter;
import com.m2mobi.markymark.InlineDisplayItem;
import com.m2mobi.markymarkcommon.markdownitems.inline.BoldString;

/**
 * Converts a {@link BoldString} to a {@link Spanned}.
 */
public class BoldInlineDisplayItem implements InlineDisplayItem<Spanned, BoldString> {

	@Override
	public Spanned create(final InlineConverter<Spanned> pInlineConverter, final BoldString pMarkDownString) {
		final Spannable spannable = SpannableUtils.createSpannable(pInlineConverter, pMarkDownString);
		spannable.setSpan(new StyleSpan(Typeface.BOLD), 0, spannable.length(), Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
		return spannable;
	}
}

File: android/src/main/java/com/m2mobi/markymarkandroid/inline/BoldInlineDisplayItem.kt
/*
* Copyright (C) M2mobi BV - All Rights Reserved
*/

package com.m2mobi.markymarkandroid.inline

import android.graphics.Typeface
import android.text.Spanned
import android.text.style.StyleSpan

import com.m2mobi.markymark.InlineConverter
import com.m2mobi.markymark.InlineDisplayItem
import com.m2mobi.markymarkcommon.markdownitems.inline.BoldString

/**
 * Converts a [BoldString] to a [Spanned].
 */
class BoldInlineDisplayItem : InlineDisplayItem<Spanned, BoldString> {

    override fun create(pInlineConverter: InlineConverter<Spanned>, pMarkdownString: BoldString): Spanned {
        val spannable = SpannableUtils.createSpannable(pInlineConverter, pMarkdownString)
        spannable.setSpan(StyleSpan(Typeface.BOLD), 0, spannable.length, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
        return spannable
    }
}